crates/forge/src/test_filter.rs

```diff
@@ -0,0 +1,454 @@
+                },
+                TestCase {
+                    name: "thing".to_string(),
+                    available_gas: None,
+                    ignored: false,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+            ]
+        );
+
+        let tests_filter = TestsFilter::new(Some("nonexistent".to_string()), false, false, false);
+        let filtered = tests_filter.filter_tests(mocked_tests.clone());
+        assert_eq!(filtered.test_cases, vec![]);
+
+        let tests_filter = TestsFilter::new(Some(String::new()), false, false, false);
+        let filtered = tests_filter.filter_tests(mocked_tests.clone());
+        assert_eq!(
+            filtered.test_cases,
+            vec![
+                TestCase {
+                    name: "crate1::do_thing".to_string(),
+                    available_gas: None,
+                    ignored: false,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+                TestCase {
+                    name: "crate2::run_other_thing".to_string(),
+                    available_gas: None,
+                    ignored: true,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+                TestCase {
+                    name: "outer::crate2::execute_next_thing".to_string(),
+                    available_gas: None,
+                    ignored: true,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+                TestCase {
+                    name: "thing".to_string(),
+                    available_gas: None,
+                    ignored: false,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+            ]
+        );
+    }
+
+    #[test]
+    fn filtering_with_no_tests() {
+        let mocked_tests = CompiledTestCrate {
+            sierra_program: program_for_testing(),
+            test_cases: vec![],
+            tests_location: CrateLocation::Lib,
+        };
+
+        let tests_filter = TestsFilter::new(Some(String::new()), false, false, false);
+        let filtered = tests_filter.filter_tests(mocked_tests.clone());
+        assert_eq!(filtered.test_cases, vec![]);
+
+        let tests_filter = TestsFilter::new(Some("thing".to_string()), false, false, false);
+        let filtered = tests_filter.filter_tests(mocked_tests.clone());
+        assert_eq!(filtered.test_cases, vec![]);
+    }
+
+    #[test]
+    fn filtering_tests_uses_whole_path() {
```

Comment:
I don't understand what this test checks, isn't name just a string so the semantic structure of path doesn't matter here?

Reply:
Not my test, I only copied it

Reply:
@cptartur

Reply:
Agree, will remove it

Reply:
It was added once we added support for filtering with strings like `crate2::`. I'd still like to test for this kind of inputs, but this can be just used as another case in normal filtering tests.

Reply:
Discusssed offline, should be removed